definitions/reports/networking.rb

```diff
@@ -0,0 +1,79 @@
+require 'socket'
+
+module Reports
+  class Networking < ForemanMaintain::Report
+    metadata do
+      description 'Report information about networking'
+    end
+
+    def run
+      subnet_counts_by_type
+      hosts_by_address_family
+      interfaces_by_address_family
+      preference_settings
+    end
+
+    private
+
+    # How many ipv4 subnets are defined in Foreman?
+    # How many ipv6 subnets are defined in Foreman?
+    def subnet_counts_by_type
+      %w[Ipv4 Ipv6].each do |type|
+        data_field("subnet_#{type.downcase}_count") do
+          sql_count("subnets where type = 'Subnet::#{type}'")
+        end
+      end
+    end
+
+    # How many hosts in Foreman have an interface with an ipv4 address but no ipv6 address?
+    # How many hosts in Foreman have an interface with no ipv4 address but an ipv6 address?
+    # How many hosts in Foreman have an interface with both ipv4 and ipv6 addresses?
+    def hosts_by_address_family
+      { 'ipv4only': 'nics.ip IS NOT NULL AND nics.ip6 IS NULL',
+        'ipv6only': 'nics.ip IS NULL AND nics.ip6 IS NOT NULL',
+        'dualstack': 'nics.ip IS NOT NULL AND nics.ip6 IS NOT NULL' }.each do |kind, condition|
+        query = <<~SQL
+          hosts
+          INNER JOIN nics on nics.host_id = hosts.id
+          WHERE #{condition}
+        SQL
+        data_field("hosts_with_#{kind}_interface_count") { sql_count(query) }
```

Comment:
If I am not mistaken, for host with two nics, you will get 2 records in the joined table.

Reply:
Yeah, you're right, fixed